src/components/products/ProductList.js

```diff
@@ -23,20 +25,70 @@ const ProductContainer = styled.div`
   gap: 25px;
   justify-content: space-evenly; ;
 `;
+const InputConatainer = styled.div`
+  display: flex;
+`;
+const Button = styled.button`
+  margin: 5px;
+  padding: 5px;
+  background: linear-gradient(to left, #1cb5e0, #000046);
+  border: 1px solid rgba(0, 0, 0, 0.2);
+  color: white;
+  border-radius: 5px;
+`;
+
+function ProductList() {
+  const [productList, setProductList] = useState(null);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [searchResults, setSearchResults] = useState([]);
+  const inputRef = useRef('');
+
+  useEffect(() => {
+    fetchProductData();
+  }, []);
+  const getinputValue = () => {
+    inputHandler(inputRef.current.value);
+  };
+
+  const fetchProductData = async () => {
+    try {
+      const productList = await axios.get('https://dummyjson.com/products');
+      setProductList(productList.data.products);
+    } catch (e) {
+      toast.error(e.message);
+    }
+  };
+  const inputHandler = (searchQuery) => {
+    setSearchQuery(searchQuery);
+
+    if (searchQuery !== '') {
+      const filteredProductList = productList.filter((listItems) => {
+        return Object.values(listItems).join('').toLocaleLowerCase().includes(searchQuery.toLocaleLowerCase());
+      });
+      setSearchResults(filteredProductList);
+    }
+  };
+  const data = searchQuery.length < 1 ? productList : searchResults;
 
-function ProductList({ data }) {
   return (
-    <ProductContainer>
-      {data?.map((items) => (
-        <Container key={items.id}>
-          <img src={items.thumbnail} alt="image loading" width="250px" height="200px" />
-          <ProductName>Model: {items.title}</ProductName>
-          <ProductName>Brand:{items.brand}</ProductName>
-          <ProductName>Discription: {items.description}</ProductName>
-          <ProductName>Ratings: {items.rating}</ProductName>
-        </Container>
-      ))}
-    </ProductContainer>
+    <>
+      <InputConatainer>
+        {' '}
+        <input value={searchQuery} onChange={getinputValue} ref={inputRef} />
+        <Button onClick={getinputValue}>Search</Button>
+      </InputConatainer>
+      <ProductContainer>
+        {data?.map((items) => (
+          <Container key={items.id}>
+            <img src={items.thumbnail} alt="image loading" width="250px" height="200px" />
+            <ProductName>Model: {items.title}</ProductName>
+            <ProductName> Brand:{items.brand}</ProductName>
```

Comment:
Why are you using ProductName styled component for everything?
I would have changed `ProductName` to `ProductInfoItem` here.
Also, creating a styled component for every html element is a bad practice. Use simple div with classes here and define them inside "Container" styled components as nested styles